### PR TITLE
use project sourceEncoding to read files

### LINKED ISSUE #3: Character encoding issues
I have a really strange bit of code I wrote a long time ago to test something.  I happened to cause a bug with this formatter where it fails to process the file.

Code to cause the error
```java
    public static void main(final String[] args) throws CharacterCodingException {
        final CharsetDecoder decoder = Charsets.ISO_8859_1.newDecoder();
        final CharsetEncoder encoder = Charsets.UTF_8.newEncoder();

        // Convert the byte array from starting inputEncoding into UCS2
        final byte[] bufferToConvert = "Hello World! £€".getBytes(Charsets.UTF_8);
        final CharBuffer cbuf = decoder.decode(ByteBuffer.wrap(bufferToConvert));

        // Convert the internal UCS2 representation into outputEncoding
        final ByteBuffer bbuf = encoder.encode(CharBuffer.wrap(cbuf));
        System.out.println(new String(bbuf.array(), 0, bbuf.limit(), Charsets.UTF_8));
    }
```

Here is the stack trace.

```
   at net.revelc.code.impsort.maven.plugin.AbstractImpSortMojo.fail (AbstractImpSortMojo.java:231)
    at net.revelc.code.impsort.maven.plugin.AbstractImpSortMojo.lambda$execute$3 (AbstractImpSortMojo.java:175)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:948)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:418)
    at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:270)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:948)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:747)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:721)
    at java.util.stream.AbstractTask.compute (AbstractTask.java:316)
    at java.util.concurrent.CountedCompleter.exec (CountedCompleter.java:731)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:289)
    at java.util.concurrent.ForkJoinPool$WorkQueue.runTask (ForkJoinPool.java:1056)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1692)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:157)
Caused by: java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException (CoderResult.java:281)
    at sun.nio.cs.StreamDecoder.implRead (StreamDecoder.java:339)
    at sun.nio.cs.StreamDecoder.read (StreamDecoder.java:178)
    at java.io.InputStreamReader.read (InputStreamReader.java:184)
    at java.io.BufferedReader.fill (BufferedReader.java:161)
    at java.io.BufferedReader.readLine (BufferedReader.java:324)
    at java.io.BufferedReader.readLine (BufferedReader.java:389)
    at java.nio.file.Files.readAllLines (Files.java:3205)
    at java.nio.file.Files.readAllLines (Files.java:3242)
    at net.revelc.code.impsort.ImpSort.parseFile (ImpSort.java:50)
    at net.revelc.code.impsort.maven.plugin.AbstractImpSortMojo.lambda$execute$3 (AbstractImpSortMojo.java:166)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:184)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:193)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:948)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:151)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:174)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:418)
    at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:270)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:948)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:481)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:471)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:747)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:721)
    at java.util.stream.AbstractTask.compute (AbstractTask.java:316)
    at java.util.concurrent.CountedCompleter.exec (CountedCompleter.java:731)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:289)
    at java.util.concurrent.ForkJoinPool$WorkQueue.runTask (ForkJoinPool.java:1056)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1692)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:157)
```

### PR DESCRIPTION
**Pull request by a14n**
Fix #3

### CODE REVIEW

File: src/test/java/net/revelc/code/impsort/ImpSortTest.java [hunk 1]
```diff
@@ -70,4 +71,12 @@ public void parseGroups() {
         Grouper.parse(" b , * , ab ,com., java , a"));
   }
 
+  @Test
+  public void testIso8859ForIssue3() throws IOException {
+    Path p =
+        Paths.get(System.getProperty("user.dir"), "src", "test", "resources", "Iso8859File.java");
+    Result result =
+        new ImpSort(StandardCharsets.ISO_8859_1.name(), eclipseDefaults, true).parseFile(p);
+    assertTrue(result.getImports().isEmpty());
```
Comment (keith-turner) [inline src/test/java/net/revelc/code/impsort/ImpSortTest.java]: could write the result out and ensure it check that it has the proper encoding

File: src/test/java/net/revelc/code/impsort/ImpSortTest.java [hunk 2]
```diff
@@ -35,14 +36,14 @@
...
-    new ImpSort(eclipseDefaults, false).parseFile(p);
+    new ImpSort(StandardCharsets.UTF_8.name(), eclipseDefaults, false).parseFile(p);
   }
 
   @Test
   public void testUnused() throws IOException {
     Path p =
         Paths.get(System.getProperty("user.dir"), "src", "test", "resources", "UnusedImports.java");
-    Result result = new ImpSort(eclipseDefaults, true).parseFile(p);
+    Result result = new ImpSort(StandardCharsets.UTF_8.name(), eclipseDefaults, true).parseFile(p);
```
Comment (keith-turner) [inline src/test/java/net/revelc/code/impsort/ImpSortTest.java]: `parseFile` could pass the encoding to Result to use when saving

File: src/test/java/net/revelc/code/impsort/ImpSortTest.java (hunk 1 above)
Comment (a14n) [reply src/test/java/net/revelc/code/impsort/ImpSortTest.java]: done

File: src/test/java/net/revelc/code/impsort/ImpSortTest.java (hunk 2 above)
Comment (a14n) [reply src/test/java/net/revelc/code/impsort/ImpSortTest.java]: done

**Comment by a14n** [comment]
PTAL

File: src/test/java/net/revelc/code/impsort/ImpSortTest.java [hunk 3]
```diff
@@ -70,4 +74,14 @@ public void parseGroups() {
...
 
+  @Test
+  public void testIso8859ForIssue3() throws IOException {
+    Path p =
+        Paths.get(System.getProperty("user.dir"), "src", "test", "resources", "Iso8859File.java");
+    Result result = new ImpSort(StandardCharsets.ISO_8859_1, eclipseDefaults, true).parseFile(p);
+    assertTrue(result.getImports().isEmpty());
+    Path output = File.createTempFile("impSort", null).toPath();
+    result.saveSorted(output);
+    assertArrayEquals(Files.readAllBytes(p), Files.readAllBytes(output));
```
[the remark this answers is not in the document]
Comment (a14n) [reply src/test/java/net/revelc/code/impsort/ImpSortTest.java]: done

**Pull request merged**